src/formatDate.js

```diff
@@ -50,7 +50,41 @@
  */
 
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const oldSeparator = fromFormat[3];
+  const newSeparator = toFormat[3];
+  const newDate = [];
+  const dateArray = date.split(oldSeparator);
+
+  for (let i = 0; i < 3; i++) {
+    let oldFormatPart = fromFormat[i];
+    let dateArrayPart = dateArray[i];
+
+    // checking years format for differences
+    if (fromFormat[i][0] !== toFormat[i][0]) {
+
+    }
+
+    if (fromFormat[i].length > toFormat[i].length
+      && fromFormat[i].slice(2) === toFormat[i]) {
+      oldFormatPart = fromFormat[i].slice(2);
+      dateArrayPart = dateArray[i].slice(2);
+    } else if (fromFormat[i].length < toFormat[i].length
+      && toFormat[i].slice(2) === fromFormat[i]) {
```

Comment:
Move logical operation to variables for more readable

Reply:
Got you, thanks